kim_commands.rst

```diff
@@ -1033,119 +1011,122 @@ property definition for more detailed information.
 
+Note that if this command is called with an instance ID that does not exist, no
+error is raised.
+
 **Modify**
 
 .. parsed-literal::
 
    kim_property modify instance_id key key_name key_name_key key_name_value
 
-*kim\_property modify* command, incrementally builds the property instance by
-receiving keys with associated arguments. Each "key\_name" is associated with
-a map containing one or more key-value pairs (in the form of
-"key\_name\_key", and "key\_name\_value" pairs).
-
-For example,
+The *kim\_property modify* command incrementally builds the property instance
+by receiving property definition keys along with associated arguments. Each
+*key\_name* is associated with a map containing one or more key-value pairs (in
+the form of *key\_name\_key*-*key\_name\_value* pairs).  For example,
 
 .. parsed-literal::
 
    kim_property modify 1 key species source-value Al
    kim_property modify 1 key mass    source-value 26.98154
+   kim_property modify 1 key mass    source-unit  amu
 
-where, a key\_name (e.g., "species" or "mass" in the above demonstration) is
-instantiated by a "key" token from the rest of key-value pairs.
-
-or,
+where the special keyword "key" is followed by a *key\_name* ("species" or
+"mass" in the above) and one or more key-value pairs.  These key-value pairs
+may continue until either another "key" keyword is given or the end of the
+command line is reached.  Thus, the above could equivalently be written as
 
 .. parsed-literal::
 
-   kim_property modify 2 key cohesive-potential-energy source-unit eV digits 5
+   kim_property modify 1 key species source-value Al       &
+                         key mass    source-value 26.98154 &
+                         key mass    source-unit  amu
 
-where, "cohesive-potential-energy" is a key\_name followed by ("source-unit",
-"eV"), and ("digits", "5") key-value pairs.
+As an example of modifying multiple key-value pairs belonging to the map of a
+single property key, the following command modifies the map of the
+"cohesive-potential-energy" property key to contain the key "source-unit" which
+is assigned a value of "eV" and the key "digits" which is assigned a value of
+5:
 
-key-value pairs with multiple keys can be modified and set one by one as
-demonstrated below,
+.. parsed-literal::
 
- .. parsed-literal::
+   kim_property modify 2 key cohesive-potential-energy source-unit eV digits 5
 
-    kim_property modify 2 key cohesive-potential-energy source-unit eV
-    kim_property modify 2 key cohesive-potential-energy digits 5
-    kim_property modify 2 key a source-unit angstrom
-    kim_property modify 2 key a digits 5
+Note that the relevant data types of the values in the map are handled
+automatically based on the `KIM Properties Framework
+<https://openkim.org/doc/schema/properties-framework>`_.  In the example above,
+this means that the value "eV" will automatically be interpreted as a string
+while the value 5 will be interpreted as an integer.
```

Comment:
again, I don't think this works as expected.

Reply:
@ellio167 would you please explain why do you think that?

Reply:
Maybe it is OK.  I was thinking that the command would not know how to correctly parse this because there could be a variable number of arguments after the 'key' argument.  However, I guess that is the old way we had things done...

I guess, now the number of arguments is just one, unless a range is given (e.g., 2:5).  So, yes, it seems that this should be OK.

If you agree, then I'm good with this as is.